lrlex/src/lib/parser.rs

```diff
@@ -201,20 +200,22 @@ impl<StorageT: TryFrom<usize>> LexParser<StorageT> {
         if declaration_parameters.is_empty() {
             return Err(self.mk_error(LexErrorKind::UnknownDeclaration, off));
         }
-        let start_states: HashSet<(&str, Span)> = declaration_parameters
+        let start_states = declaration_parameters
             .split_whitespace()
             .map(|name| {
                 let off = name.as_ptr() as usize - self.src.as_ptr() as usize;
                 let span = Span::new(off, off + name.len());
-                self.validate_start_state(span, name, errs)
+                (name, span)
             })
-            .filter_map(Result::transpose)
-            .collect::<LexInternalBuildResult<HashSet<(&str, Span)>>>()?;
+            .collect::<Vec<_>>();
 
         for (name, name_span) in start_states {
             let id = self.start_states.len();
-            let start_state = StartState::new(id, name, exclusive, name_span);
-            self.start_states.push(start_state);
+            // If false, assume an err was added to errs that will be handled later.
```

Comment:
I think the right place for this comment is a doc string on `validate_start_date` so that everyone knows what behaviour to expect from a return of `false`.

Reply:
Fixed in [58a53e7](https://github.com/softdevteam/grmtools/pull/332/commits/58a53e75f14401753958da26411b8e0fdefc233d)